tests/rules/data/syntax-error-in-prepared-statement.php

```diff
@@ -328,4 +328,13 @@ public function bug442(Connection $conn, string $table)
     {
         $conn->executeQuery("SELECT * FROM `$table`");
     }
+
+    public function bidiTest(\Dibi\Connection $conn)
+    {
+        // syntax error, should expect a error in the test
+        $conn->fetch('SELECT email adaid WHERE gesperrt FROM ada');
+        // all good, no error expected
+        $conn->fetch('SELECT email, adaid FROM ada');
+    }
```

Comment:
with the single line configuration change in `rule.neon` phpstan-dba is now able to detect this syntax error on line 335 (for mysql and postgres adapter).

I think at this point you can look into dibi and add more tests of methods which take a sql query as its arg (while also adding new entries in `rules.neon`.

Reply:
btw: looking at the dibi docs, I found examples like
> $database->query('UPDATE users SET', $arr, 'WHERE id=?', $id);

I don't think we can easily add support for these notation, where sql construction and values are intermixed

Reply:
I added the methods. The thing is that it gets complicated with the `query` method. 

`$connection->query()` returns an object on which you can call `fetch, fetchAll, fetchSingle, fetchPairs`. Or you can use the shortcut for each of these (`$connection->fetch(query)` instead of `$connection->query(query)->fetch()`)

Not really sure if this should be addressed or we should just force to use the shorter version (w/o `query()`)

> btw: looking at the dibi docs, I found examples like

yes, I was thinking that could be problematic. Perhaps this advanced syntax would not be supported at first, start with the basics

Reply:
> `$connection->query()` returns an object on which you can call `fetch, fetchAll, fetchSingle, fetchPairs`. Or you can use the shortcut for each of these (`$connection->fetch(query)` instead of `$connection->query(query)->fetch()`)

lets have a look in https://github.com/dg/dibi/blob/1585255d7d99869c313f054e1b389b45a4f5d678/src/Dibi/Connection.php#L216

we see that this method returns a object from type `Dibi\Result`. This class does not contain methods for which you can pass in a query-string, so for the goal of this PR, thats not relevant.

last step for this PR before merge: add a example similar to 
`$database->query('UPDATE users SET', $arr, 'WHERE id=?', $id);`
so we can see that `SyntaxErrorInPreparedStatementMethodRule`  does not freak out / fatal erros or creates false positives in such scenarios.

after that, this PR is good to be merged and I can give you new directions on how we can make the type inference happen

Reply:
I have added one test